### PR TITLE
Add context-dependent data structures from CVC4

### PR DESCRIPTION
**Pull request by AleksandarZeljic**
First in a series of PRs to intriduce context-dependent search implementation in Marabou. 

1. Add context-dependent (CD) data structures from CVC4 to deps folder in Marabou project. Adds all the CD data structures from CVC4, of immediate interest are Context, CDList and CDO. 

2. Fix compilation of added source files on Windows ( variadic syntax, /deps/base/check.h used some g++ specific pre-processor instructions). Noteworthy for the future, if we ever update CD files with new versions from CVC4 (due to bug fixes, updates, etc).

### CODE REVIEW

File: deps/CVC4/include/cvc4_private.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*********************                                                        */
+/*! \file cvc4_public.h
```
Comment (ahmed-irfan) [inline deps/CVC4/include/cvc4_private.h:2]: IIRC, we modified this file. It was a hack. 
Now the file header and its header guard do not match with the file name. 
Maybe, we should try to have a clean solution for this.

File: deps/CVC4/include/cvc4_private_library.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*********************                                                        */
+/*! \file cvc4_public.h
```
Comment (ahmed-irfan) [inline deps/CVC4/include/cvc4_private_library.h:2]: ditto

File: CMakeLists.txt [hunk 3]
```diff
@@ -28,6 +28,8 @@ set(MARABOU_TEST_LIB MarabouHelperTest)
 set(MARABOU_EXE Marabou${CMAKE_EXECUTABLE_SUFFIX})
 set(MARABOU_PY MarabouCore)
 
+
```
Comment (anwu1219) [inline CMakeLists.txt:31]: Extra spaces and empty lines in this file.

File: deps/CVC4/include/cvc4_private.h (hunk 1 above)
Comment (AleksandarZeljic) [reply deps/CVC4/include/cvc4_private.h]: Good catch, will fix!

File: CMakeLists.txt (hunk 3 above)
Comment (AleksandarZeljic) [reply CMakeLists.txt]: Thanks, will make spacing uniform throughout the file.

**Review by ahmed-irfan — APPROVED** [review]
LGTM.

**Pull request merged**